lib/cipherstash/index/bloom_filter.rb

```diff
@@ -0,0 +1,28 @@
+require "openssl"
+
+module CipherStash
+  class Index
+    class BloomFilter
+      attr_reader :bits
+      attr_reader :k
+
+      def initialize(key)
+        @key = key
+        @bits = Set.new()
+        @k = 3
+      end
+
+      def add(term)
+        hash = OpenSSL::HMAC.hexdigest("SHA256", @key, term)
+
+        hash.bytes[0...@k].each do |bit|
```

Comment:
I don't think that's doing what I think you think it's doing.  I won't give you the full answer straight away, in case you want to puzzle it out, but a couple of hints: check out `hash.bytes.length`, and also the distribution of values in `hash.bytes.length`, and compare them to what you'd *expect* from the output of a SHA256 hash.

Reply:
Ah right, I should be calling `OpenSSL::HMAC.digest` instead of `OpenSSL::HMAC.hexdigest`. I'll fix that up

Reply:
Fixed [here](https://github.com/cipherstash/ruby-client/pull/13/commits/67e9f2f3f88407afe938a43ac3b77492661bc181). Is that what you were thinking @mpalmer?

Reply:
Indeed.